pricegraph/src/orderbook.rs

```diff
@@ -106,10 +109,10 @@ impl Orderbook {
     /// this ring trade.
     pub fn is_overlapping(&self) -> bool {
         // NOTE: We detect negative cycles from each disconnected subgraph.
-        Subgraphs::new(self.projection.node_indices().skip(1))
+        Subgraphs::<&OrderbookGraph>::new(self.projection.node_indices().skip(1))
             .for_each_until(
-                |token| match bellman_ford::search(&self.projection, token) {
-                    Ok((_, predecessor)) => ControlFlow::Continue(predecessor),
+                |token| match ShortestPathGraph::new(&self.projection, token) {
+                    Ok(shortest_path_graph) => ControlFlow::Continue(shortest_path_graph),
                     Err(NegativeCycle(..)) => ControlFlow::Break(true),
```

Comment:
Something that was there from before that I just noticed now:
```suggestion
                    Err(_) => ControlFlow::Break(true),
```